### ISSUE TITLE
Scroll bars are missing

### ISSUE THREAD

**Issue by lassana** [overtake/TelegramSwift#195]
Steps to reproduce:

1. Go to _System Preferences_ → _General_ → _Show Scroll bars:_ → _Always_
2. Restart the Telegram app
3. Notice that the scroll bar is no longer displayed in the chat UI

Environment:
- macOS 10.13.6
- Telegram 4.2.2.135543

**Comment by patrickwelker** [comment]
Still happening with 4.7.146688 Beta and/or 4.7.146563 Stable.

- macOS 10.13.6

My system settings are _"Show scroll bars: Always"_. If I set it to _"When scrolling"_ I get my scroll bar in the Mac Telegram client.

**Comment by YARG** [comment]
> Still happening with 4.7.146688 Beta and/or 4.7.146563 Stable.
> 
> * macOS 10.13.6
> 
> My system settings are _"Show scroll bars: Always"_. If I set it to _"When scrolling"_ I get my scroll bar in the Mac Telegram client.

Yep, I have the same problem using 4.9.155352 Stable and the only way to see the scrollbars is to set to "When scrolling" as Patrick has stated above.

**Comment by alecfin** [comment]
Still here in MacOS 10.14.5 (18F203), Telegram 5.5.1 (176368)

**Comment by bratello** [comment]
Same for me, Telegram v.5.8 (185085)

**Comment by Wanny** [comment]
Same here. Using 5.8.2 beta.

**Comment by Wanny** [comment]
Update.  Still using 5.8.2 beta but another build (188892). Scrollbars **still** don't show.

I downloaded the other (non-Swift) MacOS version and scrollbars DO show in there. 
How can it be that after almost 1 and a half year since the opening of this issue is it still not addressed? :/

**Comment by bratello** [comment]
@overtake Do you have some progress with it? Thanks

**Comment by markkfjc** [comment]
On Mac OSx (Catalina), my scroll bar preferences are set to "Always". I get scroll bars in the Telegram Lite app.   Noticing the Folders are out for Desktop, I installed the non-Lite version this morning and there are no scroll bars. NADA.  I had to switch back to Telegram Lite.  Please fix the scrollbar problem.  I see this has been going on now since Aug 2018!!!

**Comment by dartess** [comment]
This problem appeared a couple of months ago.
There is no scroll. Page Up and Page Down buttons do not work.

**Comment by nguyenkien** [comment]
2021 is coming and we still don't have scrollbar

**Comment by excomdo81** [comment]
same here

**Comment by abuinteam** [comment]
Same here

**Comment by madranet** [comment]
If we plus-one this enough times do you think someone from Telegram will actually address the issue?  A few months shy of 3 years since the original issue was posted and it has not even had a reply from the devs, never mind being fixed.

C'mon Telegram! Basic functionality such as actually being able to scroll the content of the app's only window should be higher priority than adding the next cutesy animation to appeal to the emoji generation.

_<inserts "trying to scroll the window but not being able to and headbutting the monitor in sheer frustration" emoji>_

**Comment by lassana** [comment]
Moreover, if you try to revert scrollbar behavior for the Telegram app only (`defaults write ru.keepcoder.Telegram AppleShowScrollBars -string WhenScrolling`) it simply ignores it. Somehow the app always uses the system configuration value even if there is an app-specific config value set (and the app still fails to show the scrollbar)

**Comment by itsvikasgupta** [comment]
As a workaround, I am running telegram inside Bluestacks on mac.

**Comment by FrancescoSaverioZuppichini** [comment]
Same

**Comment by FantaZZ** [comment]
Same.  And page up , page down is not working

**Comment by madranet** [comment]
Now on 8,2 and still no scrollbars. 

WTF is going on Telegram? This issue is over 3 years old. Three bloody years to fix something as fundamental as being able to scroll the window content and you've still not managed it.  It's not like we can even workaround by using the arrow keys to scroll, as the up arrow is hard-wired to 'edit previous message'.

Any chance that, before the heat death of the universe, you could divert one of your developers from  those oh-so-vital tasks like adding more moronic emoji features for the illiterate and throw the rest of us a bone? 

Three years, FFS!

**Comment by LandWaterSky** [comment]
I've advocated usability issues inside companies since the inception of the Web. The biggest issue I have with basic navigation being ignored for Mac users using Telegram is that they are ignoring the HUGE number of users who are vision-impaired. In essence, people with aging eyes or those with normal vision problems. The phone app format just doesn't work for millions of these people around the world. I've almost stopped using the platform entirely because I actually fall into that group. No basic navigation on a Web interface just doesn't work.

**Comment by theirix** [comment]
Please upvote [the corresponding issue](https://bugs.telegram.org/c/3827) in the official bug tracker.

**Comment by Knocks** [comment]
This affects functionality where I can't use Telegram on a Mac anymore.  Please address it.

**Comment by madranet** [comment]
I think ---after 3 1/2 years without even an acknowledgement from the developers, much less a fix-- its safe to assume that users wanting a functional secure messenger are no longer the target demographic for Telegram. 

We should have asked for scrollbars made of dancing emojis. They'd have implemented that straight away and bragged about this great new 'feature' on their blog

**Comment by tabaquera** [comment]
I sent Telegram a message about this months ago.  Never got a response and issue still not fixed.  Come on Telegram!  At least respond to your users!

**Comment by Formartha** [comment]
Moved to web-based Telegram (e.g. via Chrome) - now it's working. that's wierd no one actually address it.

**Comment by nusic1** [comment]
Bump

**Comment by nusic1** [comment]
This should be addressed immediately

**Comment by automaciej** [comment]
Same problem here.

**Comment by elena-mrdja** [comment]
Same problem here, unbelievable

**Comment by madranet** [comment]
> I think ---after 3 1/2 years without even an acknowledgement from the developers, much less a fix-- its safe to assume that users wanting a functional secure messenger are no longer the target demographic for Telegram.

Said me, back in Feb 2022.  

Now coming up for 5 years  since this bug was first reported  [Aug 2018] and not a single response or acknowledgement from the developers. Why even bother to have a Github repo with issue tracking for your software, if you don't even look at the damned thing?

**Comment by avonar** [comment]
but.. it is still not working =(

**Comment by patrickwelker** [comment]
> but.. it is still not working =(

For what it's worth. It's working for me with build 9.7.2.250138 Stable on macOS 10.15.7.

![Screen_shot_2023-08-08_11 55 08](https://github.com/overtake/TelegramSwift/assets/591759/9d6f3d3a-bff0-44f3-a4ae-b50026a80be8)

**Comment by avonar** [comment]
i have the same version, but i dont have this preference in the app
<img width="370" alt="изображение" src="https://github.com/overtake/TelegramSwift/assets/668939/f6acc4b3-6c60-4e85-96d9-284a2a9bdd44">

**Comment by avonar** [comment]
Yes i did, there are no effects. =(

```    Click the Apple () symbol in the top-left corner of your Mac's menu bar, then click System Settings
    Using the sidebar, scroll down and select Appearance.
    Under the "Show scroll bars" option, click the radio button next to Always.```
